frontend/javascripts/admin/voxelytics/workflow_list_view.tsx

```diff
@@ -25,20 +25,24 @@ const persistence = new Persistence<Pick<{ searchQuery: string }, "searchQuery">
   "workflowList",
 );
 
-function parseRunInfo(runInfo: VoxelyticsWorkflowListingRun): VoxelyticsWorkflowListingRun {
+function parseRunInfo(runInfo: VoxelyticsWorkflowListingRun) {
   return {
     ...runInfo,
-    beginTime: new Date(runInfo.beginTime),
+    beginTime: runInfo.beginTime != null ? new Date(runInfo.beginTime) : null,
     endTime: runInfo.endTime != null ? new Date(runInfo.endTime) : null,
-  } as any as VoxelyticsWorkflowListingRun;
+  };
 }
 
 function parseWorkflowInfo(workflowInfo: VoxelyticsWorkflowListing): VoxelyticsWorkflowListing {
   return {
     ...workflowInfo,
-    runs: workflowInfo.runs
-      .map(parseRunInfo)
-      .sort((a, b) => b.beginTime.getTime() - a.beginTime.getTime()),
+    runs: workflowInfo.runs.map(parseRunInfo).sort((a, b) => {
+      if (a.beginTime != null && b.beginTime != null)
+        return b.beginTime.getTime() - a.beginTime.getTime();
+      else if (a.beginTime != null) return a.beginTime.getTime();
+      else if (b.beginTime != null) return b.beginTime.getTime();
```

Comment:
quoting from mdn on the compareFn:

> [compareFn] should return a number where:
> 
> A negative value indicates that a should come before b.
> A positive value indicates that a should come after b.
> Zero or NaN indicates that a and b are considered equal.

in line 42 and 43 you return a positive number which means that a will always come after b (regardless of which one is undefined) which doesn't really make sense. I'm not entirely sure what the context of this function is (does it sort different runs of a workflow?), but you probably want that the element with beginTime != null comes before the one with beginTime == null (or the opposite?):

```suggestion
      else if (a.beginTime != null) return 1;
      else if (b.beginTime != null) return -1;
```

or swap 1 and -1 (depending on the order you think makes sense)

Reply:
oh damn, I even read the docs but mixed it up anyways :sweat_smile: yes, it sorts the runs of a workflow once the user expanded them.

Reply:
>  but you probably want that the element with beginTime != null comes before the one with beginTime == null
exactly